caiman/base/movies.py

```diff
@@ -1408,51 +1408,62 @@ def load(file_name: Union[str, List[str]],
-            # and builds after OpenCV 4.0.1 no longer statically link that backend,
-            # instead using a separate package that invariably doesn't support as many
-            # codecs (adding gst-plugins-good doesn't help,
-            # tested up to OpenCV 4.6.0 and gst-plugins-good 1.18.5)
-            # Instead we'll explicitly go with the DirectShow backend on Windows, which
-            # seems to have good broad codec support (if this turns out to break other users,
-            # we may need a more complex solution)
+            # We first try with OpenCV, which is very dependent on having a working backend.
+            # This is backend-and-build dependent (the second argument to cv2.VideoCapture defaults to CAP_ANY, which
+            # is "the first backend that thinks it can do the job". It often works, and on Linux and OSX builds of OpenCV
+            # it usually uses GStreamer.
+            # On Windows it has used a variety of things over different releases, and if the default doesn't work, it can
+            # sometimes help to change backends (e.g. to cv2.CAP_DSHOW), but this is a guessing game. Future versions may provide
+            # a flexible route to expose this option to the caller so users don't need to tweak code to get their movies loaded.
+            #
+            # We have a fallback of trying to use the pims package if OpenCV fails
+
             cap = cv2.VideoCapture(file_name)
 
             length = int(cap.get(cv2.CAP_PROP_FRAME_COUNT))
-            width = int(cap.get(cv2.CAP_PROP_FRAME_WIDTH))
+            width  = int(cap.get(cv2.CAP_PROP_FRAME_WIDTH))
             height = int(cap.get(cv2.CAP_PROP_FRAME_HEIGHT))
 
-            cv_failed = False
             dims = [length, height, width]                     # type: ignore # a list in one block and a tuple in another
-            if length == 0 or width == 0 or height == 0:       #CV failed to load
-                cv_failed = True
-            if subindices is not None:
-                if not isinstance(subindices, list):
-                    subindices = [subindices]
-                for ind, sb in enumerate(subindices):
-                    if isinstance(sb, range):
-                        subindices[ind] = np.r_[sb]
-                        dims[ind] = subindices[ind].shape[0]
-                    elif isinstance(sb, slice):
-                        if sb.start is None:
-                            sb = slice(0, sb.stop, sb.step)
-                        if sb.stop is None:
-                            sb = slice(sb.start, dims[ind], sb.step)
-                        subindices[ind] = np.r_[sb]
-                        dims[ind] = subindices[ind].shape[0]
-                    elif isinstance(sb, np.ndarray):
-                        dims[ind] = sb.shape[0]
-
-                start_frame = subindices[0][0]
+            if length <= 0 or width <= 0 or height <= 0:       # OpenCV failure
```

Comment:
No need to do it here (not really a comment for review, per se) but I eventually intend to pull this check out into a separate function because we do it a few times in a few places